src/Symfony/Component/Messenger/Bridge/Redis/Transport/Connection.php

```diff
@@ -72,28 +75,31 @@ public function __construct(array $configuration, array $connectionCredentials =
         }
 
         $initializer = static function ($redis) use ($host, $port, $auth, $serializer, $dbIndex) {
-            $redis->connect($host, $port);
+            if (null === $redis) {
+                $redis = !\is_array($host) ? new \Redis() : new \RedisCluster(null, $host);
+            }
+
+            if ($redis instanceof \Redis) {
+                $redis->connect($host, $port);
+            }
+
             $redis->setOption(\Redis::OPT_SERIALIZER, $serializer);
 
-            if (null !== $auth && !$redis->auth($auth)) {
+            if (null !== $auth && $redis instanceof \Redis && !$redis->auth($auth)) {
                 throw new InvalidArgumentException('Redis connection failed: '.$redis->getLastError());
             }
 
-            if ($dbIndex && !$redis->select($dbIndex)) {
+            if ($dbIndex && $redis instanceof \Redis && !$redis->select($dbIndex)) {
```

Comment:
What about the dbIndex ? If it is not supported by RedisCluster, it should not be silently ignored

Reply:
[Redis Cluster doesn't support multiple databases](https://redis.io/topics/cluster-spec#implemented-subset):

>Redis Cluster does not support multiple databases like the stand alone version of Redis. There is just database 0 and the SELECT command is not allowed.

How should it fail? By throwing an `InvalidArgumentException`?

Reply:
I would say yes. If you pass a dbIndex for a cluster config, that's an InvalidArgumentException

Reply:
Fixed: https://github.com/symfony/symfony/blob/8d6de3a91e24ff63472955a45be7f9dc92a031a3/src/Symfony/Component/Messenger/Bridge/Redis/Transport/Connection.php#L94-L101